[OS-259, OS-260, OS-261, OS-262, OS-263, OS-264] Flow: Add scenes and the main app flow

tombettany opened the pull request:
Implement the core flow of the app with touchscreen scenes.

Very much a WIP but adds a basic flow that makes sense end-to-end.

tombettany committed:
Flow: Add scenes and the main app flow

Implement the core flow of the app with touchscreen scenes.

radujipa reviewed (approved):
- Digitizer is duplicated from Touch
 - Copyright headers on new files are missing in a few places
 - Note BSD license in debian/copyright for Qt
 - Make sure Qt copyright header is used where necessary

(repo is open :) )

WIP :shipit:

wizofe commented:
On @radujipa 's point, actually we ~can't cover the repo using GPL/LGPL, as it's not compatible with BSD~  Actually they 're using the modified BSD (3 clause), so that's indeed compatible!

tombettany commented:
Made changes to add the bulk of the main flow. Some tweaks are still required to pass a design review but the specification is (mostly) implemented.

To address the above issues:

 - Removed the digitizer as we aren't even using it
 - Added copyright headers to everthing
 - Rewrote files which contained the Qt BSD license

I'm going to merge in as-is so that the package isn't completely broken and follow-up with an improvement PR.
